desktop/ui/src/main/java/org/datacleaner/bootstrap/DCWindowContext.java

```diff
@@ -151,6 +153,10 @@ public void exit() {
 		}
 		if (_configuration != null) {
 			_configuration.getEnvironment().getTaskRunner().shutdown();
+			DescriptorProvider descriptorProvider = _configuration.getEnvironment().getDescriptorProvider();
+			if(descriptorProvider instanceof CompositeDescriptorProvider){
+				((CompositeDescriptorProvider) descriptorProvider).shutdown();
```

Comment:
Probably we could make this unnecesary by using daemon threads instead of normal threads. I think that would be better for other environments too since we use descriptor providers also in DC monitor, DC for Apache Spark and so on. In other words: It shouldn't be something that we have to handle specifically just in the DC desktop application.